src/main/kotlin/telegram/Handling.kt

```diff
@@ -666,81 +646,96 @@ object MafiaHandler {
                 hostInfos.update(long(0)) { gameLimit = false }
-                showHostSettings(long(1), chatId, bot)
+                showHostSettings(chatId, long(1), bot, long(0))
             }
             parametrized(shareCommand) {
                 hostInfos.get(long(0))?.let {
                     hostInfos.update(long(0)) { canShare = !it.canShare }
-                    showHostSettings(long(1), chatId, bot)
+                    showHostSettings(chatId, long(1), bot, long(0))
                 }
             }
             parametrized(canReassignCommand) {
                 hostInfos.get(long(0))?.let {
                     hostInfos.update(long(0)) { canReassign = !it.canReassign }
-                    showHostSettings(long(1), chatId, bot)
+                    showHostSettings(chatId, long(1), bot, long(0))
                 }
             }
             parametrized(deleteHostCommand) {
                 hostInfos.delete(long(0))
-                showHostSettings(long(1), chatId, bot)
+                showHostSettings(chatId, long(1), bot, int(2))
             }
             parametrized(promoteHostCommand) {
                 accounts.get(long(0))?.let {
-                    val chat = ChatId.fromId(chatId)
-                    val res = bot.sendMessage(chat, "Вы уверены, что хотите сделать ${it.fullName()} администратором?")
-                    if (res.isSuccess) {
-                        val msgId = res.get().messageId
-                        bot.editMessageReplyMarkup(
-                            chat,
-                            msgId,
-                            replyMarkup = inlineKeyboard {
-                                row {
-                                    button(confirmPromoteCommand, long(0), long(1), msgId)
-                                    button(deleteMsgCommand named "Нет", msgId)
-                                }
+                    val msgId = sendMessage(
+                        bot,
+                        chatId,
+                        "Вы уверены, что хотите сделать ${it.fullName()} администратором?"
+                    )
+                    updateMessage(
+                        bot, chatId, msgId,
+                        inlineKeyboard {
+                            row {
+                                button(confirmPromoteCommand, long(0), long(1), msgId)
+                                button(deleteMsgCommand named "Нет", msgId)
                             }
-                        )
-                    }
+                        }
+                    )
                 }
             }
             parametrized(confirmPromoteCommand) {
+                bot.deleteMessage(ChatId.fromId(chatId), long(1))
                 bot.deleteMessage(ChatId.fromId(chatId), long(2))
                 admins.save(UserId(ObjectId(), long(0)))
-                showHostSettings(long(1), chatId, bot)
+                showHostSettings(chatId, -1L, bot, long(0))
```

Comment:
Parameters been swapped around makes everything that much confusing to review, that's why I'm not even sure which parameter here does what